app/src/main/java/ch/epfl/favo/view/tabs/MapsPage.java

```diff
@@ -124,32 +137,50 @@ public View onCreateView(
     FloatingActionButton button = view.findViewById(R.id.offline_map_button);
     button.setOnClickListener(this::onOfflineMapClick);
 
+    RadioButton toggle = view.findViewById(R.id.list_switch);
+    toggle.setOnClickListener(this::onToggleClick);
+    activity = (MainActivity) Objects.requireNonNull(getActivity());
+    setupView();
+    SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+    if (mapFragment != null)
+      mapFragment.getMapAsync(this);
+    mLocation = DependencyFactory.getCurrentGpsTracker(getContext()).getLocation();
+    updateNearbyList();
     return view;
   }
 
-  @Override
-  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
-    super.onViewCreated(view, savedInstanceState);
-    setupView();
-    SupportMapFragment mapFragment =
-        (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
-    mGpsTracker = new GpsTracker(getContext());
-    if (mapFragment != null) {
-      mapFragment.getMapAsync(this);
+  private void updateNearbyList(){
+    String setting = activity.getPreferences(Context.MODE_PRIVATE).getString("radius", "1km");
+    double radius = 1;
+    switch (setting){
+      case "1 Km":
+        radius = 1; break;
+      case "5 Km":
+        radius = 5; break;
+      case "10 Km":
+        radius = 10; break;
+      case "25 Km":
+        radius = 25; break;
     }
+    CompletableFuture<List<Favor>> favors = FavorUtil.getSingleInstance()
+            .retrieveAllFavorsInGivenRadius(mLocation, radius, activity);
+    favors.whenComplete((e, res)->{
+      if (first) {
```

Comment:
Do you catch any exceptions here?

Reply:
I am gonna add this to deal with exception:     
currentFavorFuture.exceptionally(
        e -> {
          showSnackbar(getString(R.string.update_favor_error));
          return null;
        });

Reply:
I think throwing an exception here would crash the app, so we should show a snackbar.

Reply:
I changed it to:
      favors.exceptionally(
              e -> {
                CommonTools.showSnackbar(view, getString(R.string.nearby_favors_exception));
                return null;
              });